makeflow/src/makeflow_wrapper_umbrella.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (C) 2016- The University of Notre Dame
+ * This software is distributed under the GNU General Public License.
+ * See the file COPYING for details.
+ */
+
+#include <errno.h>
+#include <stdbool.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "batch_job.h"
+#include "debug.h"
+#include "dag.h"
+#include "makeflow_wrapper.h"
+#include "makeflow_wrapper_umbrella.h"
+#include "path.h"
+#include "stringtools.h"
+
+struct makeflow_wrapper_umbrella *makeflow_wrapper_umbrella_create() {
+	struct makeflow_wrapper_umbrella *w = malloc(sizeof(struct makeflow_wrapper_umbrella));
+	if(!w) {
+		fatal("malloc failed: %s\n", strerror(errno));
+	}
+
+	w->wrapper = makeflow_wrapper_create();
+	w->spec = NULL;
+	w->binary = NULL;
+	return w;
+}
+
+void makeflow_wrapper_umbrella_set_spec(struct makeflow_wrapper_umbrella *w, const char *spec) {
+	struct stat st;
+	if(lstat(spec, &st) == -1) {
+		fatal("lstat(`%s`) failed: %s\n", spec, strerror(errno));
+	}
+	if((st.st_mode & S_IFMT) != S_IFREG) {
```

Comment:
@dthain @btovar Should we start checking if the file is regular or is this overkill for Makeflow?